### PR TITLE
Quick Ban Check

### PR DESCRIPTION
**Pull request by smile0711**
This is a very simple ban check to make sure people who have a status of 0 cannot login nor stay logged in.

### CODE REVIEW

File: spa/src/components/BanCheck.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      checking: true,
+    };
+  },
+  methods: {
+    async isBanned(): Promise<any> {
+      return this.$http.post("/member/is_banned")
+        .then((response) => {
+          if (response.data.data === 0) {
+            this.$store.methods.destroySession();
+            this.$router.push({name: "login"});
```
Comment (dburleson) [inline spa/src/components/BanCheck.vue:23]: I'd suggest a new route to redirect the banned user to which states their account has been suspended. We can then use this page in the future for directions on disputes.

**Review by dburleson — CHANGES_REQUESTED** [review]
Nice work. Just a few changes.

**Comment by smile0711** [comment]
Ready for a new review

File: spa/src/main.ts [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+import App from "./App.vue";
 import api from "./api";
 import appStore from "./appStore";
 import { User } from "./appStore";
-import * as filters from './helpers/fiters';
+import * as filters from "./helpers/fiters";
 import routes from "./routes";
 import socket from "./socket";
 import "./assets/index.scss";
+import {name} from "dayjs";
```
Comment (dburleson) [inline spa/src/main.ts:13]: Why is name from dayjs being imported here?

**Review by dburleson — CHANGES_REQUESTED** [review]
Just one change to fix before I proceed to testing. Nice wording on the ban page.

File: spa/src/main.ts (hunk 2 above)
Comment (smile0711) [reply spa/src/main.ts:13]: Because my IDE doesn't mind it's business hahaha

File: api/src/services/member/member.service.ts [hunk 3]
```diff
@@ -15,7 +15,6 @@ import {
 import { Member, Place } from '../../types/models';
 import { MemberInfoView } from '../../types/views';
 import { SessionInfo } from 'session-info.interface';
-import { Request, Response } from 'express';
```
Comment (dburleson) [inline api/src/services/member/member.service.ts:18]: This lines needs to be re-added in

```
import { Request, Response } from 'express';
```

Comment (dburleson) [reply api/src/services/member/member.service.ts:18]: it also needs reverting on member.routes.ts. I accidentally suggested it to be removed in my initial review.

**Comment by smile0711** [comment]
Now everything should be good to go

**Review by dburleson — APPROVED** [review]

**Review by dburleson — APPROVED** [review]

**Pull request merged**